### ISSUE TITLE
[Content Patcher] seasonal cat no longer applied correctly

### ISSUE THREAD

**Issue by Pathoschild** [Pathoschild/LookupAnything#366]
Seasonal textures for the cat are no longer applied correctly in 1.6+, but work correctly if you downgrade to 1.5.3.

Repro steps:
1. Install the [BlueWind95 Cat Sphynx Replacer](https://www.nexusmods.com/stardewvalley/mods/2361).
2. Edit its `config.json` and set `"CatColor": "seasonal"`.
3. Launch the game.
4. Result: the cat still shows the default texture, and `patch summary` shows this:
   ```
   loaded  | conditions | applied | name + details
   ------- | ---------- | ------- | --------------
   [X]     | [X]        | [ ]     | entry #12 (Load Animals/cat)
   ```

**Comment by Pathoschild** [comment]
Content Patcher 1.6+ updates context at the `SaveLoadedBasicInfo` stage of the save load. That's after the NPCs are deserialised from the XML file (thus caching the texture), but before they're added to the game (thus letting SMAPI propagate changes). Content Patcher does re-update context when the save is fully loaded, but at that point it determines that the patch conditions haven't changed (e.g. the season in the repro steps), so it doesn't reapply them.

**Comment by Pathoschild** [comment]
Fixed in the upcoming SMAPI 2.11 via Pathoschild/SMAPI@44402c8b616fc036d5605cc29d468105cdf12d45.
